src/target/riscv/riscv.c

```diff
@@ -4375,97 +4400,155 @@ static bool gdb_regno_cacheable(enum gdb_regno regno, bool is_write)
 	}
 }
 
-/**
- * This function is called when the debug user wants to change the value of a
- * register. The new value may be cached, and may not be written until the hart
- * is resumed. */
-int riscv_set_register(struct target *target, enum gdb_regno regid, riscv_reg_t value)
+static int riscv_set_or_write_register(struct target *target,
+		enum gdb_regno regid, riscv_reg_t value, bool write_request)
```

Comment:
What does write_request mean? Looks like it's more like must_write, or write_through?

Reply:
Renamed to `write_through` and commented on.